docs/source/06_solver_hooks.rst

```diff
@@ -0,0 +1,179 @@
+    The |sdk| API function :cpp:func:`get_number_of_threads` is used to do it properly. See
+    :ref:`plugin_internal_data` section for more information.
+
+.. autofunction:: alfasim_sdk.hook_specs.finalize
+
+As can be seen in the example above the function :cpp:func:`get_plugin_data` is used to retrieved the plugin internal
+data for each `thread` identified as ``thread_id``.
+
+.. note::
+    In the examples of usage of both :py:func:`HOOK_INITIALIZE<alfasim_sdk.hook_specs.initialize>` and :py:func:`HOOK_FINALIZE<alfasim_sdk.hook_specs.finalize>`
+    there are many times where an error code is returned (``return errcode;`` or ``return OK;``). As can be seen, it is
+    possible to return error codes from |sdk| API functions, however the developer can intercept this error code and
+    handle it instead of returning it to the |alfasim|'s Solver.
+
+.. _update_secondary_variables:
+
+Update plugin variables
+-----------------------
+
+The `hooks` described in this section are related to plugin secondary variables that were registered in the python config
+file, as already explained in :ref:`solver_customization` section. They are called `secondary variables` because they
+are not obtained from |alfasim|'s Solver, these ones are called primary variables and examples of those variables are `pressure`,
+`temperature`, `volume fractions` and `velocities`.
+
+Using the following `hooks` the plugin is able to calculate/update those variables in three different moments of the
+simulation step. Two of them are called in the `Hydrodynamic Solver` scope and the last one is called in the
+`Tracers Solver` scope. Once the solver obtain results for primary variables, it updates all secondary variables in which
+depend on primary variables. After that, :py:func:`HOOK_UPDATE_PLUGINS_SECONDARY_VARIABLES<alfasim_sdk.hook_specs.update_plugins_secondary_variables>`
+is called, but if it is running the `first time step`
+:py:func:`HOOK_UPDATE_PLUGINS_SECONDARY_VARIABLES_ON_FIRST_TIMESTEP<alfasim_sdk.hook_specs.update_plugins_secondary_variables_on_first_timestep>`
+is called before. It is necessary because usually during the `first time step` some initialization tasks are needed. Then,
+if the plugin needs to initialize with some value different from ``nan``, this hook is the place to do that.
+
+.. note::
+    Different from plugin internal data, the secondary variables registered by plugins are allocated, deallocated and
+    held by |alfasim|'s Solver. It is necessary because the variables arrays are dependent on `network` with its
+    discretization and on `hydrodynamic model`, which defines the fluid flow's `phases`, `fields` and `layers`.
+
+.. autofunction:: alfasim_sdk.hook_specs.update_plugins_secondary_variables
+
+.. autofunction:: alfasim_sdk.hook_specs.update_plugins_secondary_variables_on_first_timestep
+
+The |alfasim|'s Solver is divided in two *Newton* solvers that will solve different group of equations. The first one is
```

Comment:
Please, change `Newton Solvers` to `non-linear solvers`.

Reply:
with the asterisks right?
```
*non-linear solvers*

Reply:
Yes.